### PR TITLE
Use the correct hostname in kubelet's kubeconfig

### LINKED ISSUE #22: All nodes but the first one gets "unauthorized"
I have three nodes:
- `k8s-worker-01`
- `k8s-worker-02`
- `k8s-worker-03`

When starting the kubelet on the first one, it all works good. But on the other two nodes, I get the following errors:
```
Jan 07 11:41:52 k8s-worker-02 kubelet[7251]: E0107 11:41:52.642553    7251 kubelet_node_status.go:94] Unable to register node "k8s-worker-02" with API server: nodes is forbidden: User "system:anonymous" cannot create resource "nodes" in API group "" at the cluster scope
Jan 07 11:41:52 k8s-worker-02 kubelet[7251]: E0107 11:41:52.537626    7251 kubelet.go:2266] node "k8s-worker-02" not found
Jan 07 11:41:52 k8s-worker-02 kubelet[7251]: E0107 11:41:52.637743    7251 kubelet.go:2266] node "k8s-worker-02" not found
```

... along with a bunch other, all complaining about unauthorized user "system:anonymous".

Is this something you've seen before? I've tried manually using the certificates on the nodes, and that works:

```shell
$ curl \
    --cacert /etc/kubernetes/pki/ca.pem \
    --cert /etc/kubernetes/pki/cert.pem \
    --key /etc/kubernetes/pki/key.pem \
    https://k8s-master-01.viskanint.local:6443/api/v1/nodes
```

### PR DESCRIPTION
**Pull request by anton-johansson**
Closes #22 

There was a mix of `hostvars[item].inventory_hostname` and `inventory_hostname` in a task that had `run_once: True`, causing it to give all nodes the hostname of the first node in the credentials part of kubeconfig.

Here's an example kubeconfig of my `k8s-worker-02` before this fix:
```yaml
apiVersion: v1
clusters:
- cluster:
    certificate-authority-data: <redacted>
    server: https://k8s-master-01:6443
  name: kubernetes
contexts:
- context:
    cluster: kubernetes
    user: system:node:k8s-worker-02
  name: default
current-context: default
kind: Config
preferences: {}
users:
- name: system:node:k8s-worker-01
  user:
    client-certificate-data: <redacted>
    client-key-data: <redacted>
```

### CODE REVIEW

**Comment by amimof** [comment]
The `run_once` statement was added because `kubectl config`-commands would sometimes fail due to a lock-file already being present. Kubectl only lets one process modify kubeconfigs and having multiple nodes would cause an error because it runs on `localhost`.

I suggest adding the `kubectl` role to to `masters` and `nodes` groups and let each host create their kubeconfig rather than delegating it to localhost and copying them over.

**Comment by anton-johansson** [comment]
Yeah, that sounds like a better idea, indeed!

**Comment by amimof** [comment]
That means that installing `kubectl` on localhost isn't necessary anymore and requires adding additional instructions in README. I'll make some changes and push to this branch to see how it looks.

**Comment by anton-johansson** [comment]
I haven't tried the latest commit, but now when we don't have `with_items` anymore, shouldn't `{{ hostvars[item].inventory_hostname }}`  be replaced with `{{ inventory_hostname }}`?

File: roles/kubelet/tasks/main.yml [hunk 1]
```diff
@@ -39,39 +39,16 @@
     - "{{ cluster_config_path }}/pki/master/ca.pem"
 
 - name: Create kubelet kubeconfig
-  command: "kubectl config set-cluster kubernetes --certificate-authority {{ cluster_config_path }}/pki/master/ca.pem --embed-certs=true --server=https://{{ cluster_hostname }}:{{ cluster_port }} --kubeconfig {{ cluster_config_path }}/system:node:{{ hostvars[item].inventory_hostname }}.kubeconfig"
-  delegate_to: 127.0.0.1
-  run_once: True
-  with_items: "{{ groups['nodes'] }}"
```
Comment (anton-johansson) [inline roles/kubelet/tasks/main.yml]: I haven't tried the latest commit, but now when we don't have `with_items` anymore, shouldn't `{{ hostvars[item].inventory_hostname }}`  be replaced with `{{ inventory_hostname }}`?

**Comment by amimof** [comment]
> I haven't tried the latest commit, but now when we don't have `with_items` anymore, shouldn't `{{ hostvars[item].inventory_hostname }}` be replaced with `{{ inventory_hostname }}`?

Yes it does. I will change that :)

**Comment by amimof** [comment]
@anton-johansson Have you had the chance to test the latest commit?

**Comment by anton-johansson** [comment]
No, I can do that now.

**Comment by anton-johansson** [comment]
Other than my issue in #25 (which isn't really related to this PR), this works as expected.

**Comment by amimof** [comment]
Not really, but it's necessary on localhost until i figure out how to replace it and update the README with instructions since your default `~/.kube/config` won't be ready for your new cluster after an install.

**Comment by anton-johansson** [comment]
Ah of course. But it might not be a bad idea to keep it on localhost then? Quite nice to get a ready `~/.kube/config`. :)

**Pull request merged**